core/test/element_arithmetic_test.cpp

```diff
@@ -13,16 +14,27 @@ class ElementArithmeticTest : public ::testing::Test {
   const double a = 1.2;
   const double b = 2.3;
   double val = a;
+
+  const int32_t a_int32 = int32_t(1);
+  const int64_t a_int64 = int64_t(1);
+  std::chrono::system_clock::time_point now = std::chrono::system_clock::now();
+  std::chrono::system_clock::time_point now_copy = now;
 };
 
 TEST_F(ElementArithmeticTest, plus_equals) {
   plus_equals(val, b);
   EXPECT_EQ(val, a + b);
+  EXPECT_NO_THROW(plus_equals(now, a_int32));
+  EXPECT_NO_THROW(plus_equals(now_copy, a_int64));
+  EXPECT_TRUE(now == now_copy);
 }
 
 TEST_F(ElementArithmeticTest, minus_equals) {
   minus_equals(val, b);
   EXPECT_EQ(val, a - b);
+  EXPECT_NO_THROW(plus_equals(now, a_int32));
+  EXPECT_NO_THROW(plus_equals(now_copy, a_int64));
+  EXPECT_TRUE(now == now_copy);
 }
 
```

Comment:
I think you did not add tests for your `plus` or `minus` implementation.

Reply:
Added